platforms/jvm/language-jvm/src/main/java/org/gradle/api/tasks/compile/CompileOptions.java

```diff
@@ -113,117 +93,149 @@ public CompileOptions(ObjectFactory objectFactory) {
         this.incrementalAfterFailure = objectFactory.property(Boolean.class);
         this.forkOptions = objectFactory.newInstance(ForkOptions.class);
         this.debugOptions = new DebugOptions();
+        this.getFailOnError().convention(true);
+        this.getVerbose().convention(false);
+        this.getListFiles().convention(false);
+        this.getDeprecation().convention(false);
+        this.getWarnings().convention(true);
+        this.getDebug().convention(true);
+        this.getIncremental().convention(true);
     }
 
     /**
-     * Tells whether to fail the build when compilation fails. Defaults to {@code true}.
+     * Sets whether to fail the build when compilation fails. Defaults to {@code true}.
+     *
+     * @since 8.9
      */
     @Input
-    public boolean isFailOnError() {
-        return failOnError;
-    }
+    @Incubating
+    @ReplacesEagerProperty(originalType = boolean.class)
+    public abstract Property<Boolean> getFailOnError();
 
     /**
-     * Sets whether to fail the build when compilation fails. Defaults to {@code true}.
+     * TODO: Add deprecation warning
+     *
+     * @since 8.9
      */
-    public void setFailOnError(boolean failOnError) {
-        this.failOnError = failOnError;
+    @Incubating
+    @ReplacedBy("failOnError")
+    public Property<Boolean> getIsFailOnError() {
+        return getFailOnError();
     }
```

Comment:
Since this is `@Incubating`, I think we can just drop it in 9.0, no?

Reply:
Ah, it's not already `@Incubating`, sorry, I missed that. Yes, I think we should not add `@Incubating` _and_ `@Deprecated`, instead just add the `@Deprecated`.